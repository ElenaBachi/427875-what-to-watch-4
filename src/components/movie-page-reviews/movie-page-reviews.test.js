import React from "react";
import renderer from "react-test-renderer";
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import MoviePageReviews from "./movie-page-reviews.jsx";

import NameSpace from "../../reducer/name-space.js";

const mockStore = configureStore([]);

const mock = {
  reviews: [
    {
      reviewId: 1,
      userId: 10,
      userName: `Name1`,
      rating: 8.0,
      comment: `Comment 1`,
      date: `2019-05-08T14:13:56.569Z`,
    }, {
      reviewId: 2,
      userId: 20,
      userName: `Name2`,
      rating: 9.0,
      comment: `Comment 2`,
      date: `2019-05-08T14:13:56.569Z`,
    }, {
      reviewId: 3,
      userId: 30,
      userName: `Name3`,
      rating: 10.0,
      comment: `Comment 3`,
      date: `2019-05-08T14:13:56.569Z`,
    }
  ],
};

it(`MoviePageReviews should render correctly`, () => {
  const store = mockStore({
    [NameSpace.REVIEWS]: {
      reviews: mock.reviews
    }
  });
  const tree = renderer.create(
      <Provider store={store}>
        <MoviePageReviews />
      </Provider>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
