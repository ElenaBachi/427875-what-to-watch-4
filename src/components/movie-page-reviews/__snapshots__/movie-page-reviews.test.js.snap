// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePageReviews should render correctly 1`] = `
<div
  className="movie-card__reviews movie-card__row"
>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Comment 1
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Name1
          </cite>
          <time
            className="review__date"
            dateTime={2019-05-08T14:13:56.569Z}
          >
            May 08 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        8
      </div>
    </div>
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Comment 2
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Name2
          </cite>
          <time
            className="review__date"
            dateTime={2019-05-08T14:13:56.569Z}
          >
            May 08 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        9
      </div>
    </div>
  </div>
  <div
    className="movie-card__reviews-col"
  >
    <div
      className="review"
    >
      <blockquote
        className="review__quote"
      >
        <p
          className="review__text"
        >
          Comment 3
        </p>
        <footer
          className="review__details"
        >
          <cite
            className="review__author"
          >
            Name3
          </cite>
          <time
            className="review__date"
            dateTime={2019-05-08T14:13:56.569Z}
          >
            May 08 2019
          </time>
        </footer>
      </blockquote>
      <div
        className="review__rating"
      >
        10
      </div>
    </div>
  </div>
</div>
`;
